Sources/SwiftInspectorVisitors/NestableTypeVisitor.swift

```diff
@@ -88,9 +88,18 @@ public final class NestableTypeVisitor: SyntaxVisitor {
   }
 
   public override func visit(_ node: TypealiasDeclSyntax) -> SyntaxVisitorContinueKind {
-    let typealiasVisitor = TypealiasVisitor(parentType: parentType)
-    typealiasVisitor.walk(node)
+    guard
+      !topLevelParsingTracker.hasFinishedParsing,
+      let topLevelDeclarationName = topLevelDeclaration?.nestableInfo.name
+    else {
+      assertionFailure("Encountered more than one top-level declaration. This is a usage error: a single NestableTypeVisitor instance should start walking only over a declaration syntax node")
```

Comment:
Random, but I'm starting to think if we should print these messages to std err too?

Reply:
So we can catch the issue in release builds? I like that.

Reply:
That's super smart. Maybe we can make an `assertionFailureOrLog` method that prints to STDERR in Release.

Reply:
Yeah I'm thinking something similar. Though not everyone compiles asserts out in release... but maybe we don't care about that?

Reply:
One API could be:

- If `SWIFT_OPTIMIZATION_LEVEL = -Onone` (the typical setting for debug) we assert and crash the app as normal
- If `SWIFT_OPTIMIZATION_LEVEL = -O` (typical setting for release) we log to STDERR
- If `SWIFT_OPTIMIZATION_LEVEL = -Ounchecked` we do nothing since the developer has said they want maximize efficiency

These settings are what [control](https://andybargh.com/swift-assertions/) assertions in Swift.

Then we aren't taking an explicit stance on debug or release.

Reply:
I've never done an `#if SWIFT_OPTIMIZATION_LEVEL`. Down to see if that works 🙂

Reply:
Will be a first for me too 👌